src/common/actions/create-snap.js

```diff
@@ -81,30 +81,45 @@ export function createSnap(repositoryUrl, location) { // location for tests
         .then(checkStatus)
         .then(response => {
           return response.json().then(result => {
-            if (result.status !== 'success' ||
-                result.payload.code !== 'snap-created') {
-              throw getError(response, result);
+            if (result.status !== 'success' || result.payload.code !== 'snap-created') {
+              return Promise.reject(getError(response, result));
             }
-            const startingUrl = `${BASE_URL}/${fullName}/setup`;
-            (location || window.location).href =
-              `${BASE_URL}/login/authenticate` +
-              `?starting_url=${encodeURIComponent(startingUrl)}` +
-              `&caveat_id=${encodeURIComponent(result.payload.message)}` +
-              `&repository_url=${encodeURIComponent(repositoryUrl)}`;
+
+            return Promise.resolve(result);
           });
-        })
-        .catch(error => {
-          // if LP error says there is already such snap, just redirect to builds page
-          if (error.message === 'There is already a snap package with the same name and owner.') {
-            (location || window.location).href = `${BASE_URL}/${fullName}/builds`;
-          } else {
-            dispatch(createSnapError(fullName, error));
-          }
         });
     }
   };
 }
 
+export function createSnaps(repositories, location) {
```

Comment:
This action will eventually support batch creation of snaps. This is currently blocked in the LP auth API.

Until then, I've prepared this code to accept multiple repositories, but it actually only creates a snap for the first selected repository. This allows us work on the UI without breaking the staging website.